revscores/languages/portuguese.py

```diff
@@ -1064,21 +1064,20 @@
     "zueira"
 ])
 
-class Portuguese(Language):
+def is_badword(word):
+    return STEMMER.stem(word).lower() in BADWORDS
+                
+def is_misspelled(word):
+    if word in ("a", "A"): return False
     
-    def badwords(self, words):
-        
-        for word in words:
-            
-            if STEMMER.stem(word).lower() in BADWORDS:
-                yield word
-                    
-    def misspellings(self, words):
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
         
-        for word in words:
-            
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore")
-                
-                if len(wordnet.synsets(word, lang="por")) == 0:
-                    yield word
+        return len(wordnet.synsets(word, lang="por")) == 0
+
+portuguese = Language(
+    is_badword,
+    is_misspelled
+)
+portuguese.STEMMER = STEMMER
+portuguese.BADWORDS = BADWORDS
```

Comment:
Indeed.  Upper case used with constants "often set at the module level".   https://www.python.org/dev/peps/pep-0008/#constants